### PR TITLE
Test uri

### PR DESCRIPTION
**Pull request by Chever-John**
There's one **change** I can't tell if it's needed.
`void testGenerate(List<String> hexStrings, List<List<NDEFRecord>> messages) {
  for (int i = 0; i < hexStrings.length; i++) {
    var j = i;
    if (hexStrings[j] == ' ') {
      j++;
      assert(encodeNdefMessage(messages[i]).toHexString() == hexStrings[j]);
    }
  }
}`

### CODE REVIEW

**Comment by Chever-John** [comment]
> 
`
void testGenerate(List<String> hexStrings, List<List<NDEFRecord>> messages) {
  for (int i = 0; i < hexStrings.length; i++) {
    var j = i;
    if (hexStrings[j] == ' ') {
      j++;
      assert(encodeNdefMessage(messages[i]).toHexString() == hexStrings[j]);
    }
  }
}
`

File: test/test_uri.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        UriRecord.fromString("https://github.com")
+      ],
+    ];
+
+    // Part I: test the function of ENCODE and DECODE.
+    testParse(hexStrings, messages);
+    testGenerate(hexStrings, messages);
+  });
+
+  test('ndef message with ', () {
```
Comment (Harry-Chen) [inline test/test_uri.dart:27]: 这句话没写完？

**Review by Harry-Chen — COMMENTED** [review]
除了剥离两个测试之外，还有什么工作呢？它和原来的测试对应的有什么区别呢？

File: test/test_examples_of_records/device_information_test.dart [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+          undefinedData: [
+            DataElement.fromString(255, "NFCIM"),
+          ]);
+
+      expect(() {
+        deviceInfoRecord.payload;
+      }, throwsArgumentError);
+
+      expect(() {
+        var value = [
```
Comment (Harry-Chen) [inline test/test_examples_of_records/device_information_test.dart:163]: Replace with hex string

File: test/test_examples_of_records/uri_test.dart [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    test('test the all parts of a record', () {
+      expect(UriRecord.fromString(urlName).id, equals(null));
+      expect(UriRecord.fromString(urlName).uri, equals(Uri.parse(urlName)));
+
+      expect(UriRecord.fromString(urlName).uriString, equals(urlName));
+      expect(UriRecord.fromString(urlName).iriString, equals(urlName));
+      expect(
+          UriRecord.fromString(urlName).payload,
+          equals([
```
Comment (Harry-Chen) [inline test/test_examples_of_records/uri_test.dart:38]: Replace with hex string

**Review by Harry-Chen — CHANGES_REQUESTED** [review]
Please fix your code according to my comments.

File: test/test_examples_of_records/device_information_test.dart (hunk 2 above)
Comment (Chever-John) [reply test/test_examples_of_records/device_information_test.dart:163]: Got it!

File: test/test_examples_of_records/uri_test.dart (hunk 3 above)
Comment (Chever-John) [reply test/test_examples_of_records/uri_test.dart:38]: Got it!

**Pull request closed without merging**